en/services/mission.md

```diff
@@ -235,16 +239,89 @@ In more detail, the sequence of operations is:
 1. If no `MISSION_ACK` is received the operation will eventually timeout and may be retried (see [above](#timeout)).
 
 
-### Upload Partial Mission {#upload_partial}
+### Partial Mission Upload {#upload_partial}
 
-TBD
+The diagram below shows the communication sequence to upload a partial mission to a drone (assuming all operations succeed).
 
+> **Note** The message sequence is the same as a [full mission upload](#uploading_mission) except that it is triggered with a [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) instead of a [MISSION_COUNT](../messages/common.md#MISSION_COUNT).
+  An update is expected to be robust (succeed completely or fail and revert to original version).
 
-### Download Partial Mission {#download_partial}
+<span></span>
+> **Note** Unlike for full mission update, partial update is not required to be robust; individual commands can fail.
 
-TBD
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: MISSION_WRITE_PARTIAL_LIST
+    GCS->>GCS: Start timeout
+    Drone->>GCS: MISSION_REQUEST_INT (n)
+    GCS->>GCS: Start timeout
```

Comment:
Should be the aircraft who has a timeout.

Reply:
Thanks. I also note that I have this wrong in the full upload, but I'll fix separately. I'll reverse this and add the ACK that the GCS should wait for after sending `MISSION_WRITE_PARTIAL_LIST` (at least PX4 sends in the full version).

Reply:
Actually, I'll remove that ACK, Unnecessary. I should learn to read code properly.